editor/plugins/tiles/tile_map_layer_editor.cpp

```diff
@@ -96,14 +97,14 @@ void TileMapEditorTilesPlugin::_update_toolbar() {
 	}
 }
 
-void TileMapEditorTilesPlugin::_update_transform_buttons() {
-	TileMap *tile_map = Object::cast_to<TileMap>(ObjectDB::get_instance(tile_map_id));
-	if (!tile_map) {
+void TileMapLayerEditorTilesPlugin::_update_transform_buttons() {
+	TileMapLayer *edited_layer = Object::cast_to<TileMapLayer>(ObjectDB::get_instance(edited_tile_map_layer_id));
+	if (!edited_layer) {
 		return;
 	}
 
-	Ref<TileSet> tile_set = tile_map->get_tileset();
-	if (tile_set.is_null() || selection_pattern.is_null()) {
+	Ref<TileSet> tile_set = edited_layer->get_effective_tile_set();
+	if (!tile_set.is_valid() || selection_pattern.is_null()) {
```

Comment:
Why did you change it?
```suggestion
	if (tile_set.is_null() || selection_pattern.is_null()) {
```